src/jit/objectalloc.h

```diff
@@ -52,32 +70,91 @@ inline ObjectAllocator::ObjectAllocator(Compiler* comp)
     : Phase(comp, "Allocate Objects", PHASE_ALLOCATE_OBJECTS)
     , m_IsObjectStackAllocationEnabled(false)
     , m_AnalysisDone(false)
+    , m_bitVecTraits(comp->lvaCount, comp)
+    , m_HeapLocalToStackLocalMap(comp->getAllocator())
 {
     // Disable checks since this phase runs before fgComputePreds phase.
     // Checks are not expected to pass before fgComputePreds.
-    doChecks = false;
+    doChecks                   = false;
+    m_EscapingPointers         = BitVecOps::UninitVal();
+    m_ConnGraphAdjacencyMatrix = nullptr;
 }
 
+//------------------------------------------------------------------------
+// IsObjectStackAllocationEnabled: Returns true iff object stack allocation is enabled
+//
+// Return Value:
+//    Returns true iff object stack allocation is enabled
+
 inline bool ObjectAllocator::IsObjectStackAllocationEnabled() const
 {
     return m_IsObjectStackAllocationEnabled;
 }
 
+//------------------------------------------------------------------------
+// EnableObjectStackAllocation:       Enable object stack allocation.
+
 inline void ObjectAllocator::EnableObjectStackAllocation()
 {
     m_IsObjectStackAllocationEnabled = true;
 }
 
 //------------------------------------------------------------------------
-// CanAllocateLclVarOnStack: Returns true iff local variable can not
-//                           potentially escape from the method and
-//                           can be allocated on the stack.
-inline bool ObjectAllocator::CanAllocateLclVarOnStack(unsigned int lclNum) const
+// CanAllocateLclVarOnStack: Returns true iff local variable can be
+//                           allocated on the stack.
+//
+// Arguments:
+//    lclNum   - Local variable number
+//    clsHnd   - Class handle of the variable class
+//
+// Return Value:
+//    Returns true iff local variable can be allocated on the stack.
+//
+// Notes:
+//    Stack allocation of objects with gc fields and boxed objects is currently disabled.
+
+inline bool ObjectAllocator::CanAllocateLclVarOnStack(unsigned int lclNum, CORINFO_CLASS_HANDLE clsHnd)
+{
+    assert(m_AnalysisDone);
+
+    DWORD classAttribs = comp->info.compCompHnd->getClassAttribs(clsHnd);
+
+    if ((classAttribs & CORINFO_FLG_CONTAINS_GC_PTR) != 0)
+    {
+        // TODO-ObjectStackAllocation: enable stack allocation of objects with gc fields
+        return false;
+    }
+
+    if ((classAttribs & CORINFO_FLG_VALUECLASS) != 0)
+    {
+        // TODO-ObjectStackAllocation: enable stack allocation of boxed structs
+        return false;
+    }
+
+    if (!comp->info.compCompHnd->canAllocateOnStack(clsHnd))
+    {
+        return false;
+    }
+
+    const unsigned int classSize = comp->info.compCompHnd->getHeapClassSize(clsHnd);
+
+    return !CanLclVarEscape(lclNum) && (classSize <= s_StackAllocMaxSize);
+}
+
+//------------------------------------------------------------------------
+// CanLclVarEscape:          Returns true iff local variable can
```

Comment:
Typo in "iff", same with other "ifs" in this file :)

Reply:
"iff" means "if and only if".

Reply:
oh didn't know that. nevermind then.

Reply:
https://en.wikipedia.org/wiki/If_and_only_if